README.md

```diff
@@ -0,0 +1,203 @@
+# in-toto SBoM Demo
+
+In this demo, we attempt to use in-toto to create a Software Bill of Materials
+(SBoM). Specifically, we look at the components an SBoM must contain and how
+in-toto's tooling and specification can be used to generate machine readable
+metadata with these components.
+
+For the purposes of this demo, we're defining a simple straightforward supply
+chain which includes a few traditional steps such as development and code
+review, as well as a mocked vulnerability scan that generates a JSON report.
+The metadata generated can be used to check that each step was completed by the
+authorized functionary as well as what the vulnerability scan found.
+
+### SBoM User Stories
+
+There is some consensus on what user stories a SBoM must fulfil, and that
+allows us to identify what components/information it must contain. Ideally, an
+SBoM must be able to show the provenance, pedigree and integrity of the software
+and must have the necessary components to do so.
+
+__Provenance__
+
+The SBoM must have information about the chain of custody of the software and
+identify the authors and sources of each software component.
+
+__Pedigree__
+
+The SBoM must contain information about the components that make up the software
+and can even include more in-depth information such as compilers configurations.
+
+__Integrity__
+
+The SBoM must contain cryptographic information about the software. Integrity
+also applies to the SBoM itself - it should be possible to validate the
+integrity of the SBoM metadata.
+
+in-toto's specification already allows for all of these (except for how to
+trust the delivery of in-toto metadata itself, which can be handled using TUF).
+It is therefore a matter of how in-toto is implemented in a software supply
+chain and what aspects of it is recorded. In this demo, we show that the
+existing in-toto tooling is up to the task of generating verifiable SBoMs. 
+
+### Overview of the demo supply chain
+
+We have a project owner, a developer, a code reviewer, and a vulnerability
+scanner (which generates reports ike those of JFrog Xray). Steps are defined for
+each role except the project owner who instead creates the
+[layout](https://github.com/adityasaky/in-toto-sbom-demo/blob/master/metadata/root.layout)
+for the entire supply chain. Link metadata must be generated for each step and
+signed by the authorized functionary, or `in-toto-verify` will fail to validate
+the supply chain.
+
+### Download and setup in-toto on \*NIX (Linux, OS X, ..)
+__Virtual Environments (optional)__
+
+We highly recommend to install `in-toto` and its dependencies in a
+[`virtualenv`](https://virtualenv.pypa.io/en/stable/). Just copy-paste the
+following snippet to install [`virtualenvwrapper`](https://virtualenvwrapper.readthedocs.io/en/latest/) and create a virtual environment:
```

Comment:
This second sentence may be a little confusing, considering you tell them to use `pip install virtualenvwrapper` afterwards

Reply:
The second sentence asks them to copy the following snippet which is `pip install`. Did I misunderstand you?

Also note, this section matches the main `in-toto/demo` repo so any changes will have to be perhaps made there too.

Reply:
well we probably want to change them. I'd at least hope to say "these instructions are copied over from here" rather than "install this then copy and paste this"

Reply:
Okay, gotcha, I'll clean it up and we can update in-toto/demo too.